client/src/pages/Signup.js

```diff
@@ -0,0 +1,220 @@
+  root: {
+    background: 'linear-gradient(45deg, rgba(73, 94, 203, .8) 30%, rgba(73, 145, 203, .8) 90%)',
+    color: 'white',
+    height: 48,
+    padding: '0 30px',
+    boxShadow: '0 3px 5px 2px rgba(125, 123, 135, .3)',
+    marginLeft: '10px'
+  },
+})(Button);
+
+const LoginButton = withStyles({
+  root: {
+    background: 'linear-gradient(45deg, rgba(105, 105, 105, .8) 30%, rgba(169, 169, 169, .8) 90%)',
+    color: 'white',
+    height: '5ch',
+    padding: '0 30px',
+    boxShadow: '0 3px 5px 2px rgba(125, 123, 135, .3)',
+    marginLeft: '10px'
+  },
+})(Button);
+
+const styles = {
+  root: {
+    background: 'red'
+  }
+};
+
+const Signup = () => {
+
+  const classes = useStyles();
+
+  const [formData, setFormData] = useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const [state, setState] = useState({
+    open: false,
+    vertical: 'bottom',
+    horizontal: 'center',
+    message: null
+  });
+
+  const { vertical, horizontal, open, message } = state;
+
+  const { firstName, lastName, email, password, confirmPassword } = formData;
+
+  const showAlert = (newState) => {
+    setState({ open: true, ...newState });
+  }
+
+  const handleClose = () => {
+    setState({ ...state, open: false });
+  }
+
+  const onChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  }
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    // login(email, password);
+
+    if (!areAllFieldsCompleted()) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Must fill out every field!" })
+      return
+    }
+    if (!passwordIsValid() && !emailIsValid()) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Password & E-mail are invalid!" })
+      return
+    }
+    if (!passwordIsValid() && password.length !== 0) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Password is invalid!" })
+      return
+    }
+    if (!emailIsValid() && email.length !== 0) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "E-mail is invalid!" })
+      return
+    }
+    if (areAllFieldsCompleted() && emailIsValid && !passwordsAreTheSame()) {
+      showAlert({ vertical: 'bottom', horizontal: 'center', message: "Passwords must match!" })
+      return
+    }
+  };
+
+  const passwordIsValid = () => {
+    return password.length >= 6
+  }
+
+  const areAllFieldsCompleted = () => {
+    return !(firstName.length === 0 || lastName.length === 0 || email.length === 0 || password.length === 0)
+
+  }
+
+  const emailIsValid = () => {
+    return (/^[^\s@]+@[^\s@]+\.[^\s@]+$/).test(email)
+  }
+
+  const passwordsAreTheSame = () => {
+    return password === confirmPassword
+  }
+
+  return (
+    <div className='login-sign-up-container'>
+      <img className='login-image' src={imageURL}></img>
+      <div className="sign-up-form ">
+        <div className="login-container">
```

Comment:
You may want to make a component that wraps the Signup and Login forms - it would contain the main UI of the page (the image on the left, etc.) - then you can use the children's prop so that the component can just be a wrapper around the specific Signup and Login forms. This would reduce the amount of repeated code and make it easier to manage